params/config.go

```diff
@@ -374,12 +394,27 @@ func (c *ChainConfig) checkCompatible(newcfg *ChainConfig, headHeight *big.Int,
 	if isForkIncompatible(c.MuirGlacierBlock, newcfg.MuirGlacierBlock, headHeight) {
 		return newCompatError("Muir Glacier fork block", c.MuirGlacierBlock, newcfg.MuirGlacierBlock)
 	}
+
+	// Check subnet-evm specific activations
 	if isForkIncompatible(c.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp, headTimestamp) {
 		return newCompatError("SubnetEVM fork block timestamp", c.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp)
 	}
+	currentTime := big.NewInt(time.Now().Unix())
+	if match, ni := matchNil(c.AllowListConfig, newcfg.AllowListConfig); match && !ni {
+		if isForkIncompatible(c.AllowListConfig.Timestamp(), newcfg.AllowListConfig.Timestamp(), headTimestamp) {
+			return newCompatError("AllowList fork block timestamp", c.AllowListConfig.Timestamp(), newcfg.AllowListConfig.Timestamp())
+		}
+	} else if !match {
+		return newCompatError("AllowList", currentTime, currentTime)
+	}
 
-	if isForkIncompatible(c.AllowListConfig.Timestamp(), newcfg.AllowListConfig.Timestamp(), headTimestamp) {
-		return newCompatError("AllowList fork block timestamp", c.AllowListConfig.Timestamp(), newcfg.AllowListConfig.Timestamp())
+	// Check compatibility of misc, non-timed configs ([currentTime] returns on
```

Comment:
Not sure if we want to check these here 🤷

Reply:
I think it would make sense to add a check here, but don't think using matchNil is necessary since they are the same type. Also, I think we would want to enforce a more thorough check since any change here will result in a rule set change and there's no block timestamp currently in this config to dictate when such a change would go into effect.